wrapper/src/main/java/software/amazon/jdbc/PluginServiceImpl.java

```diff
@@ -489,4 +492,38 @@ public void updateDialect(final @NonNull Connection connection) throws SQLExcept
         connection);
   }
 
+  @Override
+  public HostSpec identifyConnection(Connection connection) throws SQLException {
+    if (!(this.getDialect() instanceof TopologyAwareDatabaseCluster)) {
+      return this.getCurrentHostSpec();
```

Comment:
Probably need to return null. Provided connection may have nothing in common with the current host spec.